Added details about WaitIOCompletion error (#91134)
Azure/azure-content: articles/synapse-analytics/sql/resources-self-help-sql-on-demand.md

The error message has the following pattern:

```
Error handling external file: 'WaitIOCompletion call failed. HRESULT = ???'. File/External table name...
```

Make sure that your storage is placed in the same region as serverless SQL pool. Check the storage metrics and verify that there are no other workloads on the storage layer (uploading new files) that could saturate I/O requests.

The field HRESULT contains the result code, below are the most common error codes and potential solutions:

### [0x80070002](#tab/x80070002)

This error code means the source file is not in storage.

There are reasons why this can happen:

- The file was deleted by another application.
 - A common scenario: the query execution starts, it enumerates the files and the files are found. Later, during the query execution, a file is deleted (for example by Databricks, Spark or ADF). The query fails because the file is not found.
 - This issue can also occur with delta format. The query might succeed on retry because there is a new version of the table and the deleted file is not queried again.

- Invalid execution plan cached

>   - As a temporary mitigation, run the command ```DBCC FREEPROCCACHE```. If the problem persists create a support ticket.

Reviewer: ```suggestion
  - As a temporary mitigation, run the command `DBCC FREEPROCCACHE`. If the problem persists create a support ticket.
```
Fixing inline code formatting

### [0x80070005](#tab/x80070005)

This error can occur when the authentication method is User Identity, also known as "Azure AD pass-through" and the Azure AD access token expires.

The error message might also resemble:

```
File {path} cannot be opened because it does not exist or it is used by another process.
```

- If an Azure AD login has a connection open for more than 1 hour during query execution, any query that relies on Azure AD fails. This includes querying storage using Azure AD pass-through and statements that interact with Azure AD (like CREATE EXTERNAL PROVIDER). This affects tools that keep connections open, like in query editor in SSMS and ADS. Tools that open new connections to execute a query, like Synapse Studio, are not affected.

- Azure AD authentication token might be cached by the client applications. For example Power BI caches Azure Active Directory token and reuses the same token for one hour. The long-running queries might fail if the token expires during execution.

Consider the following mitigations:

- Restart the client application to obtain a new Azure Active Directory token.
- Consider switching to: 

>   - [Service Principal](https://docs.microsoft.com/azure/synapse-analytics/sql/develop-storage-files-storage-access-control?tabs=service-principal#supported-storage-authorization-types)
>   - [Managed identity](https://docs.microsoft.com/azure/synapse-analytics/sql/develop-storage-files-storage-access-control?tabs=managed-identity#supported-storage-authorization-types) 
>   - or [Shared access signature](https://docs.microsoft.com/azure/synapse-analytics/sql/develop-storage-files-storage-access-control?tabs=shared-access-signature#supported-storage-authorization-types)

Reviewer: ```suggestion
  - [Service Principal](develop-storage-files-storage-access-control.md?tabs=service-principal#supported-storage-authorization-types)
  - [Managed identity](develop-storage-files-storage-access-control.md?tabs=managed-identity#supported-storage-authorization-types) 
  - or [Shared access signature](develop-storage-files-storage-access-control.md?tabs=shared-access-signature#supported-storage-authorization-types)
```
Creating relative links

### [0x80070008](#tab/x80070008)

This error message can occur when the serverless SQL pool is experiencing resource contraints, or if there was a transient platform issue.

- Transient issues:
  - This error can occur when Azure detects a potential platform issue that results in a change in topology to keep the service in a healthy state.
  - This type of issue happens infrequently and is transient. Retry the query.

- High concurrency or query complexity:
  - Serverless SQL doesn't impose a maximum limit in query concurrency, it depends on the query complexity and the amount of data scanned.

>   - One serverless SQL pool can concurrently handle 1000 active sessions that are executing lightweight queries, but the numbers will drop if the queries are more complex or scan a larger amount of data. [Concurrency limits for Serverless SQL Pool](https://docs.microsoft.com/azure/synapse-analytics/sql/resources-self-help-sql-on-demand#constraints)  

Reviewer: ```suggestion
  - One serverless SQL pool can concurrently handle 1000 active sessions that are executing lightweight queries, but the numbers will drop if the queries are more complex or scan a larger amount of data. For more information, see [Concurrency limits for Serverless SQL Pool](resources-self-help-sql-on-demand.md#constraints).  
```
Relative link
